homeassistant/components/emoncms/sensor.py

```diff
@@ -150,70 +144,52 @@ def __init__(
         else:
-            self._name = name
-        self._identifier = get_id(
-            sensorid, elem["tag"], elem["name"], elem["id"], elem["userid"]
-        )
-        self._hass = hass
+            self._attr_name = name
+        self._attr_unique_id = get_id(sensorid, elem)
         self._data = data
         self._value_template = value_template
-        self._unit_of_measurement = unit_of_measurement
+        self._attr_native_unit_of_measurement = unit_of_measurement
         self._sensorid = sensorid
         self._elem = elem
 
+        self._attr_state_class = SensorStateClass.MEASUREMENT
         if unit_of_measurement in ("kWh", "Wh"):
             self._attr_device_class = SensorDeviceClass.ENERGY
             self._attr_state_class = SensorStateClass.TOTAL_INCREASING
         elif unit_of_measurement == "W":
             self._attr_device_class = SensorDeviceClass.POWER
-            self._attr_state_class = SensorStateClass.MEASUREMENT
         elif unit_of_measurement == "V":
             self._attr_device_class = SensorDeviceClass.VOLTAGE
-            self._attr_state_class = SensorStateClass.MEASUREMENT
         elif unit_of_measurement == "A":
             self._attr_device_class = SensorDeviceClass.CURRENT
-            self._attr_state_class = SensorStateClass.MEASUREMENT
         elif unit_of_measurement == "VA":
             self._attr_device_class = SensorDeviceClass.APPARENT_POWER
-            self._attr_state_class = SensorStateClass.MEASUREMENT
         elif unit_of_measurement in ("°C", "°F", "K"):
             self._attr_device_class = SensorDeviceClass.TEMPERATURE
-            self._attr_state_class = SensorStateClass.MEASUREMENT
         elif unit_of_measurement == "Hz":
             self._attr_device_class = SensorDeviceClass.FREQUENCY
-            self._attr_state_class = SensorStateClass.MEASUREMENT
         elif unit_of_measurement == "hPa":
             self._attr_device_class = SensorDeviceClass.PRESSURE
-            self._attr_state_class = SensorStateClass.MEASUREMENT
-
-        if self._value_template is not None:
-            self._state = self._value_template.render_with_possible_json_value(
-                elem["value"], STATE_UNKNOWN
-            )
-        elif elem["value"] is not None:
-            self._state = round(float(elem["value"]), DECIMALS)
-        else:
-            self._state = None
+        elif unit_of_measurement == "dB":
+            self._attr_device_class = SensorDeviceClass.SIGNAL_STRENGTH
 
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
-
-    @property
-    def native_unit_of_measurement(self):
-        """Return the unit of measurement of this entity, if any."""
-        return self._unit_of_measurement
+        self._state = elem["value"]
 
     @property
-    def native_value(self):
-        """Return the state of the device."""
-        return self._state
+    def native_value(self) -> str | float | None:
```

Comment:
You can remove this property and use `attr_native_value` directly in the `update` instead.